Banner Ads: Load from server

bjtitus opened the pull request:
| 📘 Part of: #3208 |
|:---:|

Closes [PCIOS-32](https://linear.app/a8c/issue/PCIOS-32/load-banner-ad-from-blaze-promotions-endpoint)

* Loads the promotion from server endpoint `/discover/blaze/promotions.json`
* Reuses the Discover cache handling since this endpoint is part of the Discover API
* Fixes URL handling to open http/https URLs in SafariVC and everything else with `openURL` so that deep links work
* Adds animations to the Player and Podcast list banner ads

### Animations

| Podcast List | Player |
|--|--|
| <video width=300 src="https://github.com/user-attachments/assets/28b2daad-3720-486a-953e-7cab714264b7"> | <video width=300 src="https://github.com/user-attachments/assets/9a73c306-2598-4cad-85c5-c1414bb2225f"> |


## To test

* Enable the `bannerAds` feature flag and restart
* Visit Podcasts tab
* ✅ Verify the banner ad appears
* ✅ Tap the banner ad and make sure the upsell page appears
* Open the Player
* ✅ Verify the banner ad appears
* ✅ Tap the banner ad and make sure the upsell page appears

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Add loading of Blaze Promotions

bjtitus committed:
Handle non-http links (like pktc)

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

pocketcasts commented:
<!-- REUSE_ID: e9bd6c60-948f-4b4c-bff0-5b4fa259d560 -->

Version `7.94` has now entered code-freeze, so the milestone of this PR has been updated to `7.95`.

bjtitus committed:
Improve url handling

bjtitus committed:
Add task + cancellation for banner ads

bjtitus committed:
Move Blaze promotions to DiscoverServerHandler

bjtitus committed:
Fix promotion label text

bjtitus committed:
Fix layout issues + refactor banner views

bjtitus committed:
Reload podcast list when banner ad loads

bjtitus committed:
Add animation when showing banner ad

bjtitus committed:
Add animation to Podcast List banner ad

bjtitus committed:
Switch to new Duration APIs

danielebogo reviewed (changes requested):
Hey @bjtitus thanks for fixing the animation and the margins.
There are few things though:
• From the design the `AD Pocket Casts` should be always aligned to the bottom
• Listening to ATP, the product link goes over the cover (see the left screenshot below)
| 1 | 2 |
| :-------: | :-------: |
| <img width="933" height="582" alt="Screenshot 2025-07-28 at 09 33 46" src="https://github.com/user-attachments/assets/dfe2517a-98ce-487e-94fc-ade416549932" /> | <img width="1179" height="2556" alt="IMG_0272" src="https://github.com/user-attachments/assets/38aaf00a-3b54-43b9-a828-1c3b8231cbe8" /> |

bjtitus committed:
Fix spacing in Banner Ad

bjtitus committed:
Fix Chapter Link button

bjtitus committed:
Fix swiftlint error

bjtitus commented:
Thanks for catching those! This should be ready for another look.

> The AD Pocket Casts should be always aligned to the bottom

I believe there are cases where the image is top + leading aligned and the text will flow down (especially when the text size is large). This is shown in the Themes + More tab in Figma

I fixed up the spacing for the current ad in the Player, though, so it does stay aligned to the bottom if possible.

<img width="300" alt="Simulator Screenshot - iPhone 16 - 2025-07-28 at 15 56 36" src="https://github.com/user-attachments/assets/9f818a9b-571a-4813-838c-323ed19a2f3d" />

Also fixed the chapter link button:

https://github.com/user-attachments/assets/f06e0d72-a0d6-4f3e-9a04-a1f157faa315

danielebogo reviewed (approved):
Thanks for addressing those comments :shipit:
